event_model/__init__.py

```diff
@@ -1563,13 +1629,29 @@ def _is_array(checker, instance):
 __version__ = get_versions()['version']
 del get_versions
 
-ComposeRunBundle = namedtuple('ComposeRunBundle',
-                              'start_doc compose_descriptor compose_resource '
-                              'compose_stop')
+
+@dataclass
+class ComposeRunBundle:
+    """Extensible compose run bundle. This maintains backward compatibility by unpacking into a basic
+    run bundle (start, compose_descriptor, compose_resource, stop). Further extensions are optional and
+    require keyword referencing (i.e. compose_stream_resource).
+    """
+    start_doc: dict
+    compose_descriptor: callable
+    compose_resource: callable
+    compose_stop: callable
+    compose_stream_resource: Union[callable, None] = None
```

Comment:
```suggestion
    compose_stream_resource: Optional[callable] = None
```

tiny nit, but I think it reads better and is the suggested spelling in this use case.